crates/services/executor/src/executor.rs

```diff
@@ -648,18 +667,115 @@ where
                             .events
                             .push(ExecutorEvent::MessageImported(message));
                     }
-                    Event::Transaction(_) => {
-                        // TODO: implement handling of forced transactions in a later PR
+                    Event::Transaction(relayed_tx) => {
+                        let id = relayed_tx.id();
+                        // perform basic checks
+                        let checked_tx_res = Self::validate_forced_tx(
+                            relayed_tx,
+                            header,
+                            &self.consensus_params,
+                        );
+                        // handle the result
+                        match checked_tx_res {
+                            Ok(checked_tx) => {
+                                checked_forced_txs.push(checked_tx);
+                            }
+                            Err(err) => {
+                                execution_data.events.push(
+                                    ExecutorEvent::ForcedTransactionFailed {
+                                        id,
+                                        block_height,
+                                        failure: err.to_string(),
+                                    },
+                                );
+                            }
+                        }
                     }
                 }
             }
         }
 
         execution_data.event_inbox_root = root_calculator.root().into();
 
+        Ok(checked_forced_txs)
+    }
+
+    /// Parse forced transaction payloads and perform basic checks
+    fn validate_forced_tx(
+        relayed_tx: RelayedTransaction,
+        header: &PartialBlockHeader,
+        consensus_params: &ConsensusParameters,
+    ) -> Result<CheckedTransaction, ForcedTransactionFailure> {
+        let parsed_tx = Self::parse_tx_bytes(&relayed_tx)?;
+        let checked_tx =
```

Comment:
What is the reason to put 2-3 lines of code into a separate function? It creates code duplication like below:

<img width="726" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/67982ef4-96e6-49d3-9a5f-61058950295f">

Reply:
The purpose was to clean up the code. It goes a long way when you see a function called `validate_forced_tx` to be able to immediately read, line-by-line, each of the validations that are happening and not get distracted with the implementation. I didn't do it to reduce the numbers of lines. And "is a valid variant" is separate from "the max gas is correct" when it comes to requirements, so I wrote them as separate, readable requirements.